telemetry/harvester_test.go

```diff
@@ -625,32 +625,67 @@ func TestRecordInvalidMetric(t *testing.T) {
 	}
 }
 
-func TestRequestRetryBody(t *testing.T) {
+func roundTripper(attempts int) roundTripperFunc {
 	var attempt int
-	roundTripper := roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+	return roundTripperFunc(func(req *http.Request) (*http.Response, error) {
+		defer func() { attempt++ }()
 		contentLen := int(req.ContentLength)
 		body, err := ioutil.ReadAll(req.Body)
 		if err != nil {
-			t.Fatal("error reading request body: ", err)
+			return nil, err
 		}
 		bodyLen := len(body)
 		if contentLen != bodyLen {
-			t.Errorf("content-length and body length mis-match: content=%d body=%d",
-				contentLen, bodyLen)
+			return nil, fmt.Errorf(
+				"content-length and body length mis-match: content=%d body=%d",
+				contentLen, bodyLen,
```

Comment:
This change causes the tests to pass when they shouldn't be.

Reply:
And since the benchmarks don't really need to be doing any of this content length checking, it might make sense to use two separate round trippers. Keep the test one as is, and modify this one to work just for the benchmarks. I'll leave that decision up to you though 😄